OutOfSchool/OutOfSchool.AuthCommon/Validators/CustomPasswordValidationAttribute.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System.ComponentModel.DataAnnotations;
+using Microsoft.Extensions.Localization;
+
+namespace OutOfSchool.AuthCommon.Validators;
+public class CustomPasswordValidationAttribute : ValidationAttribute
+{
+    private readonly ICustomPasswordRules passwordRules;
+
+    public CustomPasswordValidationAttribute()
+    {
+        passwordRules = new CustomPasswordRules();
+    }
+
+    public CustomPasswordValidationAttribute(ICustomPasswordRules passwordRules)
+    {
+        this.passwordRules = passwordRules;
+    }
```

Comment:
(Code style) Please avoid manual creation of class instance (i.e. you have `new CustomPasswordRules();`) as this usually introduce high coupling and breaks testability. Please use DI/Service Locator patterns instead. 

For this specific case, you need `ICustomPasswordRules` only inside the `IsValid`, and you have `ValidationContext.GetService()` that could serve your needs. Thus, there is no necessity to have private field, parameterless constructor and constructor with injection.

Reply:
Fixed. Thank you!